comp/netflow/bundle.go

```diff
@@ -0,0 +1,49 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+// Package netflow implements the "netflow" bundle, which listens for netflow
+// packets, processes them, and forwards relevant data to the backend.
+package netflow
+
+import (
+	"context"
+
+	"github.com/DataDog/datadog-agent/comp/netflow/config"
+	"github.com/DataDog/datadog-agent/comp/netflow/forwarder"
+	"github.com/DataDog/datadog-agent/comp/netflow/hostname"
+	"github.com/DataDog/datadog-agent/comp/netflow/sender"
+	"github.com/DataDog/datadog-agent/comp/netflow/server"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"go.uber.org/fx"
+)
+
+// team: network-device-monitoring
+
+// Bundle defines the fx options for this bundle.
+var Bundle = fxutil.Bundle(
+	server.Module,
+	config.Module,
+	sender.Module,
+	forwarder.Module,
+	hostname.Module,
```

Comment:
The main reason was that I wanted to use the `fx` tooling in the tests, which means we need to be able to provide everything it depends on as well; if we want to create a server in a test that uses a mock forwarder or a predefined hostname, then we need to be able to pass those in via the injector or else we can't use the injector at all.

For the hostname in particular it was also useful because the existing tests actually break if you add another test that happens to call `hostname.Get()` - it's cached globally and can't be overridden once cached, so even though we had tests that specified the hostname as part of the yaml config, this would be ignored in favor of the one fetched during a previous test. Injecting a hostname service makes it easy to provide a specific hostname per test without worrying about resetting state after each test.

That said, ideally the sender, forwarder, and hostname services wouldn't be specific to netflow. @ogaca-dd Would it be worth my proposing to add these to `core` or to provide them as part of `cmd/.../run`?

Reply:
> That said, ideally the sender, forwarder, and hostname services wouldn't be specific to netflow

For the sender and the forwarder I don't think the code is ready as the aggregator is not yet componentize (I am working on it).
For hostname I think it would be a good idea

Reply:
I will create two separate PRs - one to add the hostname service to core, and one that will create an `ndmtmp` bundle with the sender and forwarder in it (which we can delete when the sender and forwarder are fully componentized). Then I can rebase this PR on those and keep them out of this PR.

Reply:
These PRs are now merged and this has been rebased on top of them.